dawn/main/ansible/Ansible.js

```diff
@@ -1,35 +1,103 @@
 import { ipcMain } from 'electron';
 import ProtoBuf from 'protobufjs';
 import dgram from 'dgram';
+import net from 'net';
 import _ from 'lodash';
 import {
-    ansibleConnect,
-    ansibleDisconnect,
-    updateStatus,
-    updateRobotState,
+  ansibleDisconnect,
+  updateRobotState,
+  notifyChange,
+  infoPerMessage,
 } from '../../renderer/actions/InfoActions';
-import { updatePeripheral } from '../../renderer/actions/PeripheralActions';
+import { updatePeripherals } from '../../renderer/actions/PeripheralActions';
 import RendererBridge from '../RendererBridge';
 
 let runtimeIP = 'localhost';  // '192.168.128.22';
+const tcpPort = 1234;
 const clientPort = 1236; // send port
 const serverPort = 1235; // receive port
 const client = dgram.createSocket('udp4'); // sender
 const server = dgram.createSocket('udp4'); // receiver
 
+
 const dawnBuilder = ProtoBuf.loadProtoFile(`${__dirname}/ansible.proto`);
 const DawnData = dawnBuilder.build('DawnData');
 const StudentCodeStatus = DawnData.StudentCodeStatus;
 
 const runtimeBuilder = ProtoBuf.loadProtoFile(`${__dirname}/runtime.proto`);
 const RuntimeData = runtimeBuilder.build('RuntimeData');
+const notificationBuilder = ProtoBuf.loadProtoFile(`${__dirname}/notification.proto`);
+const Notification = notificationBuilder.build('Notification');
+
+let runtimeSocket = null;
+let received = false;
+
+if (!Date.now) {
+  Date.now = () => { new Date().getTime(); };
+}
+const tcp = net.createServer((sock) => {
+  console.log('Runtime Connected');
+  sock.on('end', () => {
+    console.log('Runtime Disconnected');
+  });
+  sock.on('data', (data) => {
+    received = true;
+    const decoded = Notification.decode(data);
+    if (decoded.header === Notification.Type.STUDENT_RECEIVED) {
+      RendererBridge.reduxDispatch(notifyChange(0));
```

Comment:
That's fine. Though if it's possible to also import a second copy of the proto file here, then maybe you can do that.